android/src/main/java/com/imagepicker/ImagePickerModule.java

```diff
@@ -42,566 +27,714 @@
+            currentActivity.startActivityForResult(cameraIntent, requestCode);
+        } catch (ActivityNotFoundException e) {
+            e.printStackTrace();
+        }
     }
 
-    if (options.hasKey("customButtons")) {
-      ReadableMap buttons = options.getMap("customButtons");
-      ReadableMapKeySetIterator it = buttons.keySetIterator();
-      // Keep the current size as the iterator returns the keys in the reverse order they are defined
-      int currentIndex = titles.size();
-      while (it.hasNextKey()) {
-        String key = it.nextKey();
-
-        titles.add(currentIndex, key);
-        actions.add(currentIndex, buttons.getString(key));
-      }
+    // NOTE: Currently not reentrant / doesn't support concurrent requests
+    @ReactMethod
+    public void launchImageLibrary(final ReadableMap options, final Callback callback) {
+        mResponse = Arguments.createMap();
+        Activity currentActivity = getCurrentActivity();
+
+        if (currentActivity == null) {
+            mResponse.putString("error", "can't find current Activity");
+            callback.invoke(mResponse);
+            return;
+        }
+
+        parseOptions(options);
+
+        Intent libraryIntent;
+        int requestCode;
+
+        if (mPickVideo) {
+            requestCode = REQUEST_LAUNCH_VIDEO_LIBRARY;
+            libraryIntent = new Intent(Intent.ACTION_PICK);
+            libraryIntent.setType("video/*");
+        } else {
+            requestCode = REQUEST_LAUNCH_IMAGE_LIBRARY;
+            libraryIntent = new Intent(Intent.ACTION_PICK,
+                    android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+
+            mImageURI = null;
+            if (mAllowEditing) {
+                File imageFile = createOutputFile();
+                if (imageFile == null) {
+                    mResponse.putString("error", "Cannot create file");
+                    callback.invoke(mResponse);
+                    return;
+                }
+
+                mImageURI = Uri.fromFile(imageFile);
+                libraryIntent.putExtra(MediaStore.EXTRA_OUTPUT, mImageURI);
+                libraryIntent.putExtra("crop", "true");
+                libraryIntent.putExtra("aspectX", mAspectX);
+                libraryIntent.putExtra("aspectY", mAspectY);
+            }
+        }
+
+        if (libraryIntent.resolveActivity(mReactContext.getPackageManager()) == null) {
+            mResponse.putString("error", "Cannot launch photo library");
+            callback.invoke(mResponse);
+            return;
+        }
+
+        try {
+            mCallback = callback;
+            currentActivity.startActivityForResult(libraryIntent, requestCode);
+        } catch (ActivityNotFoundException e) {
+            e.printStackTrace();
+        }
     }
 
-    titles.add(cancelButtonTitle);
-    actions.add("cancel");
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (isValidResult(requestCode)) {
```

Comment:
My big mistake... Function body was doing something opposite of its' name... thnx